src/neuroconv/datainterfaces/behavior/neuralynx/neuralynx_nvt_interface.py

```diff
@@ -0,0 +1,112 @@
+import json
+from typing import Optional
+
+import numpy as np
+from pynwb import NWBFile
+from pynwb.behavior import CompassDirection, Position, SpatialSeries
+
+from .nvt_utils import read_data, read_header
+from ....basetemporalalignmentinterface import BaseTemporalAlignmentInterface
+from ....utils import DeepDict, FilePathType, NWBMetaDataEncoder
+
+
+class NeuralynxNvtInterface(BaseTemporalAlignmentInterface):
+    """Data interface for Neuralynx NVT files. NVT files store position tracking information"""
+
+    def __init__(self, file_path: FilePathType, verbose: bool = True):
+        """
+        Interface for writing Neuralynx .nvt files to nwb.
+
+        Parameters
+        ----------
+        file_path : FilePathType
+            Path to the .nvt file
+        verbose : bool, default: True
+            controls verbosity.
+        """
+
+        self.file_path = file_path
+        self.verbose = verbose
+        self._timestamps = self.get_original_timestamps()
+        self.header = read_header(self.file_path)
+        super().__init__(file_path=file_path)
+
+    def get_original_timestamps(self) -> np.ndarray:
+        data = read_data(self.file_path)
+
+        times = data["TimeStamp"] / 1000000  # Neuralynx stores times in microseconds
+        times = times - times[0]
+
+        return times
+
+    def get_timestamps(self) -> np.ndarray:
+        return self._timestamps
+
+    def set_aligned_timestamps(self, aligned_timestamps: np.ndarray) -> None:
+        self._timestamps = aligned_timestamps
+
+    def get_metadata(self) -> DeepDict:
+        metadata = super().get_metadata()
+        metadata["NWBFile"].update(session_start_time=self.header["TimeCreated"])
+        return metadata
+
+    def add_to_nwbfile(
+        self,
+        nwbfile: NWBFile,
+        metadata: Optional[dict] = None,
+        add_position: bool = True,
+        add_angle: Optional[bool] = None,
+    ):
+        """
+        Add NVT data to a given in-memory NWB file
+
+        Parameters
+        ----------
+        nwbfile : NWBFile
+            nwb file to which the recording information is to be added
+        metadata : dict, optional
+            metadata info for constructing the nwb file.
+        add_position : bool, default=True
+        add_angle : bool, optional
+            If None, write angle as long as it is not all 0s
+        """
+
+        data = read_data(self.file_path)
+
+        if add_position:
+            # convert to float and change <= 0 (null) to NaN
+            xi = data["Xloc"]
+            x = xi.astype(float)
+            x[xi <= 0] = np.nan
+
+            yi = data["Yloc"]
+            y = yi.astype(float)
+            y[yi <= 0] = np.nan
+
+            spatial_series = SpatialSeries(
+                name="NvtSpatialSeries",
+                data=np.c_[x, y],
+                reference_frame="unknown",
```

Comment:
I imagine reference frame is something that the person who set up the rig might know; should we expose the ability to specify this?